code/modules/power/engines/tesla/energy_ball.dm

```diff
@@ -104,22 +114,33 @@
 	if(length(orbiting_balls))
 		. += "There are [length(orbiting_balls)] mini-balls orbiting it."
 
-/obj/singularity/energy_ball/proc/move_the_basket_ball(move_amount)
-	var/list/dirs = GLOB.alldirs.Copy()
-	if(length(shocked_things))
-		for(var/i in 1 to 30)
-			var/atom/real_thing = pick(shocked_things)
-			dirs += get_dir(src, real_thing) //Carry some momentum yeah? Just a bit tho
-	for(var/i in 0 to move_amount)
-		var/move_dir = pick(dirs) //ensures teslas don't just sit around
-		if(target && prob(10))
-			move_dir = get_dir(src,target)
-		var/turf/T = get_step(src, move_dir)
+/obj/singularity/energy_ball/proc/move_the_basket_ball(where_to_move)
+	if(locate(/obj/machinery/field/containment) in urange(15, src, 1)) // If there's a field in the closest 15 turfs we're sticking with a simple moving pattern
+		var/turf/T = get_step(src, pick(GLOB.alldirs))
 		if(can_move(T))
 			forceMove(T)
-			setDir(move_dir)
-			for(var/mob/living/carbon/C in loc)
-				dust_mobs(C)
+			return
+		return
+	if(!has_a_target)
+		find_the_basket()
+		return
+	for(var/i in 0 to 8)
+		forceMove(tesla_line[1])
+		tesla_line.Cut(1, 2)
+		if(get_turf(src) == target_turf)
+			has_a_target = FALSE
+		for(var/mob/living/carbon/C in loc)
+			dust_mobs(C)
+
+
+/obj/singularity/energy_ball/proc/find_the_basket()
+	var/area/where_to_move = findUnrestrictedEventArea() // Grabs a random area that isn't restricted
```

Comment:
I'd rather this be a static list of valid areas generated when the tesla is created or something. Running the code in `findUnrestrictedEventArea()` every 2 seconds is unnecessary.

Reply:
It isn't ran every 2 seconds, it's only ran if the tesla has no target, or if it has reached it's destination

Reply:
Ok fair

Reply:
Sorry, that was a bit too harsh and short of an answer. I'll test around with it tonight. My only fear is that something weird could happen if a room suddenly gets deleted from the station, but is still on the list, and it then tried to move to that room.

Reply:
You can if you want, but you don't have to.